packages/office-ui-fabric-react/src/components/FloatingPicker/BaseFloatingPicker.tsx

```diff
@@ -16,19 +16,21 @@ export interface IBaseFloatingPickerState {
   didBind: boolean;
 }
 
-export class BaseFloatingPicker<T, P extends IBaseFloatingPickerProps<T>> extends BaseComponent<P, IBaseFloatingPickerState>
+// TODO the TProps here is not needed when bidning in a renderProp.
+export class BaseFloatingPicker<TItem, TProps extends IBaseFloatingPickerProps<TItem> = IBaseFloatingPickerProps<TItem>>
+  extends BaseComponent<TProps, IBaseFloatingPickerState>
   implements IBaseFloatingPicker {
   protected selection: Selection;
 
   protected root = React.createRef<HTMLDivElement>();
-  protected suggestionStore: SuggestionsStore<T>;
-  protected suggestionsControl: SuggestionsControl<T>;
-  protected SuggestionsControlOfProperType: new (props: ISuggestionsControlProps<T>) => SuggestionsControl<T> = SuggestionsControl as new (
-    props: ISuggestionsControlProps<T>
-  ) => SuggestionsControl<T>;
-  protected currentPromise: PromiseLike<T[]>;
-
-  constructor(basePickerProps: P) {
+  protected suggestionStore: SuggestionsStore<TItem>;
+  protected suggestionsControl: SuggestionsControl<TItem>;
+  protected SuggestionsControlOfProperType: new (props: ISuggestionsControlProps<TItem>) => SuggestionsControl<
+    TItem
+  > = SuggestionsControl as new (props: ISuggestionsControlProps<TItem>) => SuggestionsControl<TItem>;
```

Comment:
Seeing casts always makes me a bit nervous. Is it still needed here? (I noticed you were able to remove a similar cast above in the `BasePickerFloating` test.)